--- boost_1_61_0/boost/detail/utf8_codecvt_facet.hpp.orig	2016-05-05 23:11:21.000000000 +0200
+++ boost_1_61_0/boost/detail/utf8_codecvt_facet.hpp	2016-05-14 10:06:25.157699390 +0200
@@ -113,7 +113,7 @@
     public std::codecvt<wchar_t, char, std::mbstate_t>  
 {
 public:
-    BOOST_UTF8_DECL explicit utf8_codecvt_facet(std::size_t no_locale_manage=0)
+    explicit utf8_codecvt_facet(std::size_t no_locale_manage=0)
         : std::codecvt<wchar_t, char, std::mbstate_t>(no_locale_manage) 
     {}
     virtual  ~utf8_codecvt_facet(){}
--- boost_1_61_0/libs/serialization/build/Jamfile.v2.orig	2016-05-05 23:14:17.000000000 +0200
+++ boost_1_61_0/libs/serialization/build/Jamfile.v2	2016-05-15 07:59:08.364394013 +0200
@@ -80,7 +80,6 @@
     xml_iarchive
     xml_oarchive
     xml_archive_exception
-    codecvt_null
     utf8_codecvt_facet
     singleton
 ;
@@ -93,6 +92,7 @@
     xml_wgrammar
     xml_wiarchive
     xml_woarchive
+    codecvt_null
 ;
 
 lib boost_serialization 
