--- boost_1_35_0/tools/build/v2/util/os.jam.org	2007-06-16 15:29:24.717971888 +0200
+++ boost_1_35_0/tools/build/v2/util/os.jam	2007-06-16 15:31:19.139516164 +0200
@@ -129,22 +129,7 @@
 # Returns true if running on windows, whether in cygwin or not.
 rule on-windows ( )
 {
-    local result ;
-    if [ modules.peek : NT ]
-    {
-        result = true ;
-    }
-    else if [ modules.peek : UNIX ]
-    {
-        switch [ modules.peek : JAMUNAME ]
-        {
-            case CYGWIN* :
-            {
-                result = true ;
-            }
-        }
-    }
-    return $(result) ;
+    return true ;
 }
 
 
--- boost_1_37_0/tools/build/v2/tools/gcc.jam.orig	2008-10-16 10:42:03.000000000 +0200
+++ boost_1_37_0/tools/build/v2/tools/gcc.jam	2008-11-06 20:59:27.405556275 +0100
@@ -292,7 +292,7 @@
 
 # Declare flags and action for compilation.
 toolset.flags gcc.compile OPTIONS <optimization>off   : -O0 ;
-toolset.flags gcc.compile OPTIONS <optimization>speed : -O3 ;
+toolset.flags gcc.compile OPTIONS <optimization>speed : ;
 toolset.flags gcc.compile OPTIONS <optimization>space : -Os ;
 
 toolset.flags gcc.compile OPTIONS <inlining>off  : -fno-inline ;
--- boost_1_37_0/tools/build/v2/tools/types/lib.jam.orig	2008-10-16 10:42:03.000000000 +0200
+++ boost_1_37_0/tools/build/v2/tools/types/lib.jam	2008-11-06 21:01:59.897548732 +0100
@@ -9,13 +9,13 @@
 # platforms.
 type.register STATIC_LIB : a lib : LIB ;
 
-type.set-generated-target-suffix STATIC_LIB : <target-os>windows : lib ;
+type.set-generated-target-suffix STATIC_LIB : <target-os>windows : a ;
 type.set-generated-target-suffix STATIC_LIB : <target-os>cygwin : lib ;
 type.set-generated-target-prefix STATIC_LIB : : lib ;
 
 type IMPORT_LIB : : STATIC_LIB ;
-type.set-generated-target-suffix IMPORT_LIB : : lib ;
-type.set-generated-target-prefix IMPORT_LIB : : "" ;
+type.set-generated-target-suffix IMPORT_LIB : : dll.a ;
+type.set-generated-target-prefix IMPORT_LIB : : lib ;
 
 type.register SHARED_LIB : so dll dylib : LIB ;
 type.set-generated-target-suffix SHARED_LIB : <target-os>windows : dll ;
--- boost_1_36_0/libs/math/build/Jamfile.v2.org	2008-08-22 14:35:44.643893747 +0000
+++ boost_1_36_0/libs/math/build/Jamfile.v2	2008-08-22 14:36:11.767894179 +0000
@@ -16,7 +16,6 @@
 fmax
 fmin
 fpclassify
-hypot
 lgamma
 llround
 log1p
--- boost_1_37_0/libs/serialization/build/Jamfile.v2.orig	2008-10-07 07:34:46.000000000 +0200
+++ boost_1_37_0/libs/serialization/build/Jamfile.v2	2008-11-06 21:02:41.401548732 +0100
@@ -80,19 +80,6 @@
     xml_oarchive
 ;
     
-WSOURCES = 
-    basic_text_wiprimitive
-    basic_text_woprimitive
-    text_wiarchive
-    text_woarchive
-    utf8_codecvt_facet
-    xml_wgrammar
-    xml_wiarchive
-    xml_woarchive
-    codecvt_null
-;
-
-
 lib boost_serialization 
     : $(SOURCES).cpp
     : 
@@ -100,11 +87,4 @@
     <link>shared:<define>BOOST_SERIALIZATION_DYN_LINK=1
     ;
 
-lib boost_wserialization 
-    : $(WSOURCES).cpp boost_serialization 
-    :     
-    <toolset>msvc:<cxxflags>/Gy 
-    <link>shared:<define>BOOST_SERIALIZATION_DYN_LINK=1
-    ;
-
-boost-install boost_serialization boost_wserialization ;
+boost-install boost_serialization ;
--- boost_1_37_0/tools/build/v2/tools/types/lib.jam.orig	2008-11-06 23:30:46.449560465 +0100
+++ boost_1_37_0/tools/build/v2/tools/types/lib.jam	2008-11-07 06:59:38.737552084 +0100
